### DISCUSSION TITLE
How does Prisma know when to fetch data?

### DISCUSSION THREAD

**Discussion by vim-diesel** [prisma/prisma#24729] [Q&A] [answered]
### Question

I am curious how this works. Using NextJS, I have a table displaying rows of data, and I use query parameters, and each time the page loads, it makes a call to prisma using `findMany` on a server component with pagination and sorting. When I spam the sorting, the url switches back and forth:
```
http://localhost:3000/events?page=2&orderBy=title&direction=asc
...
http://localhost:3000/events?page=2&orderBy=title&direction=desc
...
```
and I check the log, ( I am using a Prisma Client extension to alter the findMany() query to log to console) and even though I spammed the direction change it only fetched the data twice:

```
fetching data with prisma:  { orderBy: { title: 'desc' }, take: 10, skip: 10 }
fetching data with prisma:  { orderBy: { title: 'asc' }, take: 10, skip: 10 }
```

So there is some kind of caching going on, but I don't know where, or how long, or when it decides to fetch from the cache. It seems to work great, but like any good developer I want to at least be aware of the features Prisma is using when I call `findMany()`. 

Can anyone please show me where in the docs I might find such info? Thank you

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @vim-diesel 👋 

Thank you for raising this question.

The caching behavior you are experiencing when using Prisma with Next.js is likely due to the Next.js Data Cache, which caches the results of fetch requests by default. Since Prisma uses fetch internally, this can lead to the behavior you're observing. To opt out of the Next.js Data Cache, you can [disable caching for a specific route](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic) using the "force-dynamic" config option so that up to date data is always returned and no caching happens. Do take a look at this [section](https://www.prisma.io/docs/orm/more/help-and-troubleshooting/help-articles/nextjs-prisma-client-dynamic#problem) of the documentation.

**Reply by vim-diesel** [reply]
thank you!

**Reply by ludralph** [reply]
You're welcome.

**Reply by ludralph** [reply]
Hi @vim-diesel, could you please let us know if you found it easy to locate this information in the documentation before starting this discussion?
